control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -2862,6 +2871,78 @@ func (r *reconciler) reconcileAzureCloudNodeManager(ctx context.Context, image s
 	return errs
 }
 
+// reconcileGlobalPullSecret reconciles the original pull secret given by HCP and merges it with a new pull secret provided by the user.
+// The new pull secret is only stored in the DataPlane side so, it's not exposed in the API. It lives in the kube-system namespace of the DataPlane.
+// If that PS exists, the HCCO deploys a DaemonSet which mounts the whole Root FS of the node, and merges the new PS with the original one.
+// If the PS doesn't exist, the HCCO doesn't do anything.
+func (r *reconciler) reconcileGlobalPullSecret(ctx context.Context, hcp *hyperv1.HostedControlPlane, cpoImage string) error {
+	var (
+		userProvidedPullSecretBytes []byte
+		originalPullSecretBytes     []byte
+		globalPullSecretBytes       []byte
+		err                         error
+	)
+	log := ctrl.LoggerFrom(ctx)
+
+	// Get the user provided pull secret
+	userProvidedPullSecret := manifests.UserProvidedPullSecret()
+	if err := r.uncachedClient.Get(ctx, client.ObjectKeyFromObject(userProvidedPullSecret), userProvidedPullSecret); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
```

Comment:
What happens if user deleted the `userProvidedPullSecret` after the `globalpullsecret` has been synced to the worker nodes ? Will it be restored back to `originalPullSecret` ?